src/RA_AchievementPopup.cpp

```diff
@@ -139,43 +146,52 @@ void AchievementPopup::Render(HDC hDC, RECT& rcDest)
         nDescX += 64 + 4;
     }
 
-    const std::string sTitle = std::string(" " + ActiveMessage().Title() + " ");
-    const std::string sSubTitle = std::string(" " + ActiveMessage().Subtitle() + " ");
+    const auto sTitle{ " " + ActiveMessage().Title() + " " };
+    const auto sSubTitle{ " " + ActiveMessage().Subtitle() + " " };
+
+    SelectFont(hDC, hFontTitle.get());
+    TextOut(hDC, nTitleX, nTitleY, NativeStr(sTitle).c_str(), ra::strlen_as_int(sTitle));
 
-    SelectObject(hDC, hFontTitle);
-    TextOut(hDC, nTitleX, nTitleY, NativeStr(sTitle).c_str(), sTitle.length());
-    SIZE szTitle = { 0, 0 };
-    GetTextExtentPoint32(hDC, NativeStr(sTitle).c_str(), sTitle.length(), &szTitle);
+    auto lpszTitle{ std::make_unique<SIZE>() };
+    GetTextExtentPoint32(hDC, NativeStr(sTitle).c_str(), ra::strlen_as_int(sTitle), lpszTitle.get());
 
-    SIZE szAchievement = { 0, 0 };
-    if (ActiveMessage().Subtitle().length() > 0)
+    auto lpszAchievement{ std::make_unique<SIZE>() };
```

Comment:
Alright, with my findings it seems objects that need long lifetimes need to be allocated on the heap (like globals) and objects with very short lifetimes should be allocated on the stack. I reaffirmed that if something needs a large amount of memory allocated, it should be on the heap.

According the Microsoft (since this is for Windows currently) stack size is 1 KB in kernel and 16 KB in user mode. What should we define as "too large".